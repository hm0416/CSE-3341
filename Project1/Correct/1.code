program
	A(ref r) begin r=0; endfunc
	ref x;
begin
	x = new;
	x=1;
	output 4;
end