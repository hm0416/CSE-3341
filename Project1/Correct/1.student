program
	int x;
begin
	x=10;
end
