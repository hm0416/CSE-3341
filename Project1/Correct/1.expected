PROGRAM
INT
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
BEGIN
INPUT
ID[X]
SEMICOLON
ID[Y]
ASSIGN
CONST[0]
SEMICOLON
INPUT
ID[XY]
COMMA
ID[Y]
SEMICOLON
OUTPUT
ID[XY]
SEMICOLON
ID[Y]
ASSIGN
ID[Y]
ADD
ID[XY]
SEMICOLON
ID[X]
ASSIGN
ID[X]
SUB
CONST[1]
SEMICOLON
WHILE
CONST[0]
LESS
ID[X]
BEGIN
INPUT
ID[XY]
SEMICOLON
OUTPUT
ID[XY]
SEMICOLON
ID[Y]
ASSIGN
ID[Y]
ADD
ID[XY]
SEMICOLON
ID[X]
ASSIGN
ID[X]
SUB
CONST[1]
SEMICOLON
ENDWHILE
SEMICOLON
OUTPUT
ID[Y]
SEMICOLON
END
